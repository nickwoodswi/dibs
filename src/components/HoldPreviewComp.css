.holdPreview{
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
}
#delete-button{
    margin-top: 5px;
    height: 36px;
    background-color: white;
}