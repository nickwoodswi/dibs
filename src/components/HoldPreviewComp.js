import React, { Component } from 'react'
import './HoldPreviewComp.css'

class HoldPreviewComp extends Component {
    render() {
        return(
            <div className="holdPreview">
                {this.props.startMonthPrev}<> </> 
                {this.props.startDayPrev},<> </> 
                {this.props.startYearPrev}<> </>  - <> </> 
                {this.props.endMonthPrev}<> </> 
                {this.props.endDayPrev},<> </> 
                {this.props.endYearPrev}<> </> 
                <button id="delete-button" onClick={() => this.props.deleteHold(
                    this.props.startMonthPrev, 
                    this.props.startDayPrev,
                    this.props.startYearPrev,
                    this.props.endMonthPrev,
                    this.props.endDayPrev,
                    this.props.endYearPrev,
                )}>REMOVE</button>
            </div>
        )
    }
}

export default HoldPreviewComp