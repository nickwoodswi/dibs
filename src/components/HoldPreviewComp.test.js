import React from 'react';
import { render } from '@testing-library/react';
import HoldPreviewComp from './HoldPreviewComp';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders individual hold preview component', () => {
    render(<MemoryRouter><HoldPreviewComp /></MemoryRouter>);
});

it('HoldPreviewComp renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><HoldPreviewComp /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});