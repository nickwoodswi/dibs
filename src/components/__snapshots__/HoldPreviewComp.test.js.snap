// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoldPreviewComp renders the UI as expected 1`] = `
<div
  className="holdPreview"
>
   
  ,
   
   
    - 
   
   
  ,
   
   
  <button
    id="delete-button"
    onClick={[Function]}
  >
    REMOVE
  </button>
</div>
`;
